### DISCUSSION TITLE
How are the jreleaser artifacts linked to the builds

### DISCUSSION THREAD

**Discussion by yschimke** [jreleaser/jreleaser#628] [Ideas]
Firstly, my understanding is that even in gradle config, the builds are loosely associated with the publishing steps via path names.

> Artifacts refer to a specific file as they have a single path property.

So running jreleaserFullRelease without running a build before will fail missing artifacts, unless you do something like

```
  val nativeImageDist = tasks["graalDistZip"]
  tasks["jreleaserAssemble"].dependsOn(nativeImageDist)
```

This doesn't appear to work, maybe because of lifecycle issues that I'm not a gradle expert enough to fix.

Should this link be supported when you configure using gradle?

**Comment by aalmiray** [comment]
This can be tricky. JReleaser used to have a direct dependency on outputs created by the default `distZip` and `disTar` tasks, which collided with custom distribution tasks (such as is the case when building GraalVM Native Image archive like you do). For this reason we took out the feature.

In a single project build it would be easy to say `jreleaserRelease` or any task that requires artifacts to have a dependency on `assemble` for example. On a multi-project build it's more difficult as the root project may not have all the information, or JReleaser was configured in a subproject.

For this reason it's better to explicitly invoke the task you need, in this case likely `gradle assemble jreleaserFullRelease`.

**Reply by yschimke** [reply]
OK, I'll probably try to get the dependsOn wiring working.

**Comment by gotson** [comment]
I stumbled upon the same issue. Why can't gradle find the jreleaser tasks when doing `tasks.findByName("jreleaserAssemble")` ?

I have declared some [files](https://jreleaser.org/guide/latest/reference/files.html) in JReleaser, i would expect the relevant Gradle tasks within JReleaser to register some input files for those path, is that something that could be done ?

**Reply by aalmiray** [reply]
I believe the issue is tasks applied by the JReleaser plugin are registered in an afterEvaluate block because by then the model can be safely resolved.

Perhaps this should change, eagerly registering tasks.

**Reply by gotson** [reply]
> Perhaps this should change, eagerly registering tasks.

yes, so we could use gradle dependency model :) Do you want me to create an issue for that?

**Reply by aalmiray** [reply]
Filed #1152
